File: languages/erlang/exercises/concept/booleans/booleans.erl
-module(booleans).

%%% Implement the function can_fast_attack/1
-spec can_fast_attack(KnightIsAwake :: boolean()) -> boolean().

%%% Implement the function can_spy/3
-spec can_spy(KnightIsAwake :: boolean(),
              ArcherIsAwake :: boolean(),
              PrisonerIsAwake :: boolean()) -> boolean().

%%% Implement the function can_signal_prisoner/2
-spec can_signal_prisoner(ArcherIsAwake :: boolean(),
                          PrisonerIsAwake :: boolean()) -> boolean().

%%% Implement the function can_free_prisoner/4
-spec can_free_prisoner(KnightIsAwake :: boolean(),
                        ArcherIsAwake :: boolean(),
                        PrisonerIsAwake :: boolean(),
                        PetDogIsPresent :: boolean()) -> boolean().

File: languages/javascript/exercises/concept/booleans/booleans.js
/**
 * If the knight is sleeping, then the fast attack is not available.
 *
 * @param {boolean} knightIsAwake
 * 
 * @return {boolean} Whether or not you can execute a fast attack.
 */
export function canExecuteFastAttack(knightIsAwake) {
  throw new Error('Implement the canExecuteFastAttack function')
}

/**
 * You can only spy if someone is awake; you wouldn't want to go through all the trouble just to find out nothing's going to happen, would you?
 *
 * @param {boolean} knightIsAwake
 * @param {boolean} archerIsAwake
 * @param {boolean} prisonerIsAwake
 * 
 * @returns {boolean} Whether or not you can spy on someone.
 */
export function canSpy(knightIsAwake, archerIsAwake, prisonerIsAwake) {
  throw new Error('Implement the canSpy function')
}

export function canSignalPrisoner(archerIsAwake, prisonerIsAwake) {
  throw new Error('Implement the canSignalPrisoner function')
}

export function canFreePrisoner(knightIsAwake, archerIsAwake, prisonerIsAwake, petDogIsPresent) {
  throw new Error('Implement the canFreePrisoner function')
}

